File: day5/exercises-solutions/hello.c
#include <mpi.h>
#include <stdio.h>

int main(int argc, char **argv)
{
    int size,rank;
    MPI_Init(&argc, &argv);
    MPI_Comm_size(MPI_COMM_WORLD,&size);
    MPI_Comm_rank(MPI_COMM_WORLD,&rank);

    printf("Hello, World! I am rank %d "
           "out of a total of %d tasks\n", rank, size);

    MPI_Finalize();
    return 0;
}

File: day5/exercises-solutions/hello.f90
PROGRAM hello
  USE mpi
  IMPLICIT NONE
  INTEGER :: size, rank, ierr

  CALL MPI_INIT(ierr)
  CALL MPI_COMM_SIZE(MPI_COMM_WORLD,size,ierr)
  CALL MPI_COMM_RANK(MPI_COMM_WORLD,rank,ierr)
  
  PRINT*,"Hello, World! I am rank ",rank," out of a total of",size," tasks"

  CALL MPI_FINALIZE(ierr)
END PROGRAM hello
